sdk/servicebus/service-bus/src/core/messageSender.ts

```diff
@@ -250,11 +256,46 @@ export class MessageSender extends LinkEntity {
    * @param sendBatch Boolean indicating whether the encoded message represents a batch of messages or not
    * @return {Promise<Delivery>} Promise<Delivery>
    */
-  private _trySend(encodedMessage: Buffer, sendBatch?: boolean): Promise<void> {
-    const retryOptions = this._getSenderOptions?.retryOptions || {};
-    retryOptions.timeoutInMs = getRetryAttemptTimeoutInMs(retryOptions);
+  private _trySend(
+    encodedMessage: Buffer,
+    sendBatch: boolean,
+    options?: OperationOptions
+  ): Promise<void> {
+    const abortSignal = options?.abortSignal;
+    const retryTimeoutInMs =
+      this._getSenderOptions?.retryOptions?.timeoutInMs || Constants.defaultOperationTimeoutInMs;
```

Comment:
So they are not allowed to specify 0?

Or should '||' be "??"?

Reply:
Well, we already call `getRetryAttemptTimeoutInMs()` which ensures that the `timeoutInMs` is a number. So `||` here is not needed, but TS compiler wouldn't like that.

But, this method here doesn't know that `timeoutInMs` is guaranteed to be a number by this point, so am ok with using ?? here.

Fixed with https://github.com/Azure/azure-sdk-for-js/pull/8159/commits/6cbdf74a7113093ff8998dda15665f4c440aab21

Reply:
On a side note, I am not too keen on what `getRetryAttemptTimeoutInMs()` is actually doing, so logged #8193 for further investigations